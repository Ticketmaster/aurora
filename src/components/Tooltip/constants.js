export const SPACE_FROM_MOUSE = 20;

export const directions = {
  top: "top",
  bottom: "bottom",
  left: "left",
  right: "right"
};

export const variants = {
  dark: "dark",
  light: "light"
};
