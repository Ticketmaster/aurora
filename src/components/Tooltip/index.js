import React, { Component } from "react";
import PropTypes from "prop-types";
import { CSSTransition } from "react-transition-group";
import Portal from "../PopOver/PopOverPortal";
import PopOver from "../PopOver/index";
import { StyledTooltip } from "./Tooltip.styles";
import SPACE_FROM_MOUSE from "./constants";
import {
  DIRECTIONS,
  TOP,
  LEFT,
  RIGHT,
  BOTTOM,
  VARIANTS,
  LIGHT,
  AUTO,
  ARROW_WIDTH
} from "../constants";
import { constants } from "../../theme";

class Tooltip extends Component {
  constructor(props) {
    super(props);

    this.myRef = React.createRef();
    this.dimensions = {
      width: 0,
      height: 0,
      windowScroll: 0,
      windowWidth: 0,
      windowHeight: 0
    };

    this.pos = {
      x: 0,
      y: 0
    };

    this.actualDirection = props.direction;

    this.state = {
      actualDirection: props.direction,
      arrowAdjustment: 0
    };
  }

  /*
   * Static function that needs to be called from the parent -> Tooltip.getDimensionsFromEvent
   * The parent should pass the click event which will trigger showing the Tooltip.
   * By default the Tooltip is shown withing the view port. If we need to show it inside
   * certain element we should pass the html element as second parameter.
   * The function will return an object that should be provided to the Tooltip as props.
   */
  static getDimensionsFromEvent(e, parent) {
    const {
      y: elTop,
      height: elHeight,
      x: elLeft,
      width: elWidth
    } = e.target.getBoundingClientRect();
    const {
      y: offsetTop = 0,
      height: clientHeight = 100000,
      x: offsetLeft = 0,
      width: clientWidth = 100000
    } =
      parent && parent.getBoundingClientRect
        ? parent.getBoundingClientRect()
        : {};

    return {
      elTop,
      elLeft,
      elWidth,
      elBottom: elTop + elHeight,
      elRight: elLeft + elWidth,
      offsetTop,
      clientHeight,
      offsetLeft,
      clientWidth,
      elHorizontalCenter: elLeft + elWidth / 2,
      elVerticalCenter: elTop + elHeight / 2
    };
  }

  getPositionAndUpdateDirection = ({
    position,
    dimensions,
    spaceFromMouse,
    reduce
  }) => {
    const result = PopOver.calculatePosition({
      position,
      dimensions,
      spaceFromMouse,
      reduce
    });
    const { actualDirection, arrowAdjustment } = this.state;

    const adjustment = this.adjustArrow({
      coords: { x: result.x, width: dimensions.width },
      position
    });

    if (
      result.y < position.elTop + dimensions.windowScroll &&
      (actualDirection !== TOP || arrowAdjustment !== adjustment)
    ) {
      this.actualDirection = TOP;
      this.setState({ actualDirection: TOP, arrowAdjustment: adjustment });
    } else if (
      result.y > position.elTop + dimensions.windowScroll &&
      (actualDirection !== BOTTOM || arrowAdjustment !== adjustment)
    ) {
      this.actualDirection = BOTTOM;
      this.setState({ actualDirection: BOTTOM, arrowAdjustment: adjustment });
    }

    return result;
  };

  getTranslateByDirection = direction => {
    switch (direction) {
      case TOP:
        return "translate(0, 10px)";
      case BOTTOM:
        return "translate(0, -10px)";
      case LEFT:
        return "translate(10px, 0)";
      case RIGHT:
      default:
        return "translate(-10px, 0)";
    }
  };

  adjustArrow = ({ coords, position }) => {
    const reqCenter = position.elHorizontalCenter;
    const currentCenter = coords.x + coords.width / 2;
    const mostLeft = coords.x + ARROW_WIDTH / 2;
    const mostRight = coords.x + coords.width + ARROW_WIDTH / 2;
    const center = Math.min(Math.max(mostLeft, reqCenter), mostRight);

    return center - currentCenter;
  };

  /*
   * Function that determines Tooltip position.
   * @position(object) - top and bottom position of the elemt that triggers showing Tooltip;
   * Mouse horizontal position on the sreen - so we can center the Tooltip;
   * Additional container position and size
   * @size(object) - Tooltip width and height;
   */
  calculatePosition = ({
    direction,
    position,
    dimensions,
    spaceFromMouse,
    reduce
  }) => {
    const { width, height, windowScroll } = dimensions;
    const { elBottom, elTop, elLeft, elRight, elHorizontalCenter } = position;

    const bottomPosition = elBottom + SPACE_FROM_MOUSE;
    const topPosition = elTop - SPACE_FROM_MOUSE - height + windowScroll;

    switch (direction) {
      case TOP:
        return {
          x: elHorizontalCenter - width / 2,
          y: topPosition
        };
      case BOTTOM:
        return {
          x: elHorizontalCenter - width / 2,
          y: bottomPosition + windowScroll
        };
      case LEFT:
        return {
          x: elLeft - width - SPACE_FROM_MOUSE,
          y: elTop + windowScroll
        };
      case RIGHT:
        return {
          x: elRight + SPACE_FROM_MOUSE,
          y: elTop + windowScroll
        };
      default:
        return this.getPositionAndUpdateDirection({
          direction,
          position,
          dimensions,
          spaceFromMouse,
          reduce
        });
    }
  };

  updateSize = () => {
    const { isVisible } = this.props;
    const {
      windowScroll,
      windowWidth,
      windowHeight,
      width,
      height
    } = this.dimensions;
    const dimensions = {};
    if (global.window && isVisible) {
      const {
        clientWidth,
        clientHeight
      } = global.window.document.documentElement;

      const scrollTop = Math.max(
        global.window.pageYOffset,
        global.document.documentElement.scrollTop,
        global.document.body.scrollTop
      );

      if (scrollTop !== windowScroll) {
        dimensions.windowScroll = scrollTop;
      }

      if (clientWidth !== windowWidth) {
        dimensions.windowWidth = clientWidth;
      }

      if (clientHeight !== windowHeight) {
        dimensions.windowHeight = clientHeight;
      }
    }

    if (this.myRef.current) {
      const { clientWidth, clientHeight } = this.myRef.current;

      if (width !== clientWidth && clientWidth) {
        dimensions.width = clientWidth;
      }

      if (height !== clientHeight && clientHeight) {
        dimensions.height = clientHeight;
      }
    }

    if (Object.keys(dimensions).length) {
      this.dimensions = {
        ...this.dimensions,
        ...dimensions
      };
      return true;
    }

    return false;
  };

  tooltipEnter = () => {
    const {
      isVisible,
      position,
      direction,
      spaceFromMouse,
      reduceTop,
      reduceBottom
    } = this.props;

    if (isVisible) {
      const reduce = { top: reduceTop, bottom: reduceBottom };
      this.updateSize();
      this.pos = this.calculatePosition({
        direction,
        position,
        dimensions: this.dimensions,
        spaceFromMouse,
        reduce
      });
    }

    this.myRef.current.style.top = `${this.pos.y}px`;
    this.myRef.current.style.left = `${this.pos.x}px`;
    this.myRef.current.style.transition = `opacity 0.3s ${
      constants.easing.easeOutQuad
    }`;
    this.myRef.current.style.transform = this.getTranslateByDirection(
      this.actualDirection
    );
  };

  tooltipEntering = () => {
    this.myRef.current.style.transition = `opacity 0.3s ${
      constants.easing.easeOutQuad
    },
      transform 0.3s ${constants.easing.easeOutQuad}`;
    this.myRef.current.style.transform = "translate(0)";
  };

  tooltipExit = () => {
    this.myRef.current.style.transition = `opacity 0.1s ${
      constants.easing.easeInQuad
    }`;
  };

  render() {
    const { children, isVisible, variant, ...rest } = this.props;
    const { actualDirection, arrowAdjustment } = this.state;

    return (
      <Portal>
        <CSSTransition
          in={isVisible}
          key="tooltip-animation"
          timeout={300}
          classNames="open"
          onEnter={this.tooltipEnter}
          onEntering={this.tooltipEntering}
          onExit={this.tooltipExit}
          appear={isVisible}
          variant={variant}
        >
          <StyledTooltip
            ref={this.myRef}
            isVisible={isVisible}
            {...rest}
            direction={actualDirection}
            arrowAdjustment={`${arrowAdjustment}px`}
          >
            {children}
          </StyledTooltip>
        </CSSTransition>
      </Portal>
    );
  }
}

Tooltip.propTypes = {
  children: PropTypes.node.isRequired,
  isVisible: PropTypes.bool,
  direction: PropTypes.oneOf(DIRECTIONS),
  position: PropTypes.shape({
    elHorizontalCenter: PropTypes.number,
    elVerticalCenter: PropTypes.number,
    elTop: PropTypes.number,
    elBottom: PropTypes.number,
    elLeft: PropTypes.number
  }),
  variant: PropTypes.oneOf(VARIANTS),
  spaceFromMouse: PropTypes.number,
  reduceTop: PropTypes.number,
  reduceBottom: PropTypes.number
};

Tooltip.defaultProps = {
  isVisible: false,
  direction: AUTO,
  variant: LIGHT,
  position: {
    mouseX: 0,
    elTop: 0,
    elBottom: 0,
    offsetTop: 0,
    clientHeight: 0,
    offsetLeft: 0,
    clientWidth: 0
  },
  spaceFromMouse: SPACE_FROM_MOUSE,
  reduceTop: 0,
  reduceBottom: 0
};

Tooltip.displayName = "Tooltip";

export default Tooltip;
