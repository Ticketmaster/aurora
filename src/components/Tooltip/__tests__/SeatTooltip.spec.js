import React from "react";
import renderer from "react-test-renderer";

import Tooltip from "../index";
import SeatTooltip from "../SeatTooltip";

jest.mock("../../PopOver/PopOverPortal", () => ({ children }) => children);

jest.mock("react-transition-group", () => ({
  CSSTransition: ({ children }) => children
}));

describe("SeatTooltip", () => {
  it("should match snapshot when no children and dark", () => {
    const tree = renderer
      .create(<SeatTooltip row={5} section="" seat={25} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
  it("should match snapshot when small and dark", () => {
    const tree = renderer
      .create(
        <SeatTooltip size="small" variant="dark" row={5} section="B" seat={25}>
          <div>test</div>
        </SeatTooltip>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should match snapshot when has children", () => {
    const tree = renderer
      .create(
        <SeatTooltip row={5} section="B" seat={25}>
          <div>test</div>
        </SeatTooltip>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("getDimensionsFromEvent should call Tooltip.getDimensionsFromEvent and directly return the result", () => {
    const mockGetDimensions = jest.fn(() => ({ x: 5, y: 10 }));
    Tooltip.getDimensionsFromEvent = mockGetDimensions;

    SeatTooltip.getDimensionsFromEvent("hello");

    expect(mockGetDimensions).toHaveBeenCalledTimes(1);
    expect(mockGetDimensions).toHaveBeenCalledWith("hello", undefined);
    expect(SeatTooltip.getDimensionsFromEvent("hello")).toEqual({
      x: 5,
      y: 10
    });
  });
});
