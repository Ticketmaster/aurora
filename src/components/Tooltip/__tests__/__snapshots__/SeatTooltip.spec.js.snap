// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeatTooltip should match snapshot when has children 1`] = `
.c0 {
  background-color: rgba(255,255,255,1);
  border: 1px solid #999999;
  border-radius: 4px;
  box-shadow: 0 4px 4px 0 rgba(0,0,0,0.12);
  position: absolute;
  max-width: 260px;
  color: #262626;
  padding: 8px;
  display: none;
  font-size: 12px;
  -webkit-transition: opacity 0.1s cubic-bezier(0.55,0.085,0.68,0.53), -webkit-transform 0.1s cubic-bezier(0.55,0.085,0.68,0.53);
  -webkit-transition: opacity 0.1s cubic-bezier(0.55,0.085,0.68,0.53), transform 0.1s cubic-bezier(0.55,0.085,0.68,0.53);
  transition: opacity 0.1s cubic-bezier(0.55,0.085,0.68,0.53), transform 0.1s cubic-bezier(0.55,0.085,0.68,0.53);
}

.c0:before {
  content: "";
  position: absolute;
  -webkit-transition: opacity 0.1s cubic-bezier(0.55,0.085,0.68,0.53), scale 0.1s cubic-bezier(0.55,0.085,0.68,0.53);
  transition: opacity 0.1s cubic-bezier(0.55,0.085,0.68,0.53), scale 0.1s cubic-bezier(0.55,0.085,0.68,0.53);
  display: none;
  border-right: 1px solid #999999;
  border-bottom: 1px solid #999999;
  border-top-left-radius: 100%;
  width: 12px;
  height: 12px;
  -webkit-transform: translateY(-50%) rotate(-135deg);
  -ms-transform: translateY(-50%) rotate(-135deg);
  transform: translateY(-50%) rotate(-135deg);
  background-color: rgba(255,255,255,1);
  left: calc(50% - 6px);
  top: -1px;
  -webkit-transform: translateY(-50%) rotate(-135deg);
  -ms-transform: translateY(-50%) rotate(-135deg);
  transform: translateY(-50%) rotate(-135deg);
}

.c0.open-enter,
.c0.open-enter:before {
  -webkit-transition: opacity 0.3s cubic-bezier(0.455,0.03,0.515,0.955);
  transition: opacity 0.3s cubic-bezier(0.455,0.03,0.515,0.955);
  display: block;
  opacity: 0;
}

.c0.open-enter {
  -webkit-transition: opacity 0.3s cubic-bezier(0.455,0.03,0.515,0.955), -webkit-transform 0.3s cubic-bezier(0.455,0.03,0.515,0.955);
  -webkit-transition: opacity 0.3s cubic-bezier(0.455,0.03,0.515,0.955), transform 0.3s cubic-bezier(0.455,0.03,0.515,0.955);
  transition: opacity 0.3s cubic-bezier(0.455,0.03,0.515,0.955), transform 0.3s cubic-bezier(0.455,0.03,0.515,0.955);
  -webkit-transform: scale(0.7);
  -ms-transform: scale(0.7);
  transform: scale(0.7);
}

.c0.open-enter-active,
.c0.open-enter-active:before {
  -webkit-transition: opacity 0.3s cubic-bezier(0.455,0.03,0.515,0.955);
  transition: opacity 0.3s cubic-bezier(0.455,0.03,0.515,0.955);
  display: block;
  opacity: 1;
}

.c0.open-enter-active {
  -webkit-transition: opacity 0.3s cubic-bezier(0.455,0.03,0.515,0.955), -webkit-transform 0.3s cubic-bezier(0.455,0.03,0.515,0.955);
  -webkit-transition: opacity 0.3s cubic-bezier(0.455,0.03,0.515,0.955), transform 0.3s cubic-bezier(0.455,0.03,0.515,0.955);
  transition: opacity 0.3s cubic-bezier(0.455,0.03,0.515,0.955), transform 0.3s cubic-bezier(0.455,0.03,0.515,0.955);
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.c0.open-enter-done,
.c0.open-enter-active:before {
  -webkit-transition: opacity 0.3s cubic-bezier(0.455,0.03,0.515,0.955);
  transition: opacity 0.3s cubic-bezier(0.455,0.03,0.515,0.955);
  display: block;
  opacity: 1;
}

.c0.open-enter-done {
  -webkit-transition: opacity 0.3s cubic-bezier(0.455,0.03,0.515,0.955), -webkit-transform 0.3s cubic-bezier(0.455,0.03,0.515,0.955);
  -webkit-transition: opacity 0.3s cubic-bezier(0.455,0.03,0.515,0.955), transform 0.3s cubic-bezier(0.455,0.03,0.515,0.955);
  transition: opacity 0.3s cubic-bezier(0.455,0.03,0.515,0.955), transform 0.3s cubic-bezier(0.455,0.03,0.515,0.955);
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.c0.open-exit,
.c0.open-exit:before {
  display: block;
  opacity: 1;
}

.c0.open-exit {
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.c0.open-exit-active,
.c0.open-exit-active:before {
  display: block;
  opacity: 0;
}

.c0.open-exit-active {
  -webkit-transform: scale(0.7);
  -ms-transform: scale(0.7);
  transform: scale(0.7);
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 16px;
  border-bottom: 1px solid #999999;
  line-height: 1;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  text-align: center;
}

.c2:first-child {
  text-align: left;
}

.c2:last-child {
  text-align: right;
}

.c3 {
  font-size: 14px;
  text-transform: uppercase;
  margin-bottom: 8px;
  color: #999999;
}

.c4 {
  font-size: 16px;
  font-weight: 600;
  color: #262626;
}

.c5 {
  padding: 16px;
  font-size: 14px;
  color: #262626;
}

<div
  className="c0"
  direction="bottom"
  style={
    Object {
      "padding": 0,
      "width": "260px",
    }
  }
>
  <div
    className="c1"
    size="large"
  >
    <div
      className="c2"
    >
      <div
        className="c3"
        size="large"
      >
        Sec
      </div>
      <div
        className="c4"
        size="large"
      >
        B
      </div>
    </div>
    <div
      className="c2"
    >
      <div
        className="c3"
        size="large"
      >
        Row
      </div>
      <div
        className="c4"
        size="large"
      >
        5
      </div>
    </div>
    <div
      className="c2"
    >
      <div
        className="c3"
        size="large"
      >
        Seat
      </div>
      <div
        className="c4"
        size="large"
      >
        25
      </div>
    </div>
  </div>
  <div
    className="c5"
    size="large"
  >
    <div>
      test
    </div>
  </div>
</div>
`;

exports[`SeatTooltip should match snapshot when no children and dark 1`] = `
.c0 {
  background-color: rgba(255,255,255,1);
  border: 1px solid #999999;
  border-radius: 4px;
  box-shadow: 0 4px 4px 0 rgba(0,0,0,0.12);
  position: absolute;
  max-width: 260px;
  color: #262626;
  padding: 8px;
  display: none;
  font-size: 12px;
  -webkit-transition: opacity 0.1s cubic-bezier(0.55,0.085,0.68,0.53), -webkit-transform 0.1s cubic-bezier(0.55,0.085,0.68,0.53);
  -webkit-transition: opacity 0.1s cubic-bezier(0.55,0.085,0.68,0.53), transform 0.1s cubic-bezier(0.55,0.085,0.68,0.53);
  transition: opacity 0.1s cubic-bezier(0.55,0.085,0.68,0.53), transform 0.1s cubic-bezier(0.55,0.085,0.68,0.53);
}

.c0:before {
  content: "";
  position: absolute;
  -webkit-transition: opacity 0.1s cubic-bezier(0.55,0.085,0.68,0.53), scale 0.1s cubic-bezier(0.55,0.085,0.68,0.53);
  transition: opacity 0.1s cubic-bezier(0.55,0.085,0.68,0.53), scale 0.1s cubic-bezier(0.55,0.085,0.68,0.53);
  display: none;
  border-right: 1px solid #999999;
  border-bottom: 1px solid #999999;
  border-top-left-radius: 100%;
  width: 12px;
  height: 12px;
  -webkit-transform: translateY(-50%) rotate(-135deg);
  -ms-transform: translateY(-50%) rotate(-135deg);
  transform: translateY(-50%) rotate(-135deg);
  background-color: rgba(255,255,255,1);
  left: calc(50% - 6px);
  top: -1px;
  -webkit-transform: translateY(-50%) rotate(-135deg);
  -ms-transform: translateY(-50%) rotate(-135deg);
  transform: translateY(-50%) rotate(-135deg);
}

.c0.open-enter,
.c0.open-enter:before {
  -webkit-transition: opacity 0.3s cubic-bezier(0.455,0.03,0.515,0.955);
  transition: opacity 0.3s cubic-bezier(0.455,0.03,0.515,0.955);
  display: block;
  opacity: 0;
}

.c0.open-enter {
  -webkit-transition: opacity 0.3s cubic-bezier(0.455,0.03,0.515,0.955), -webkit-transform 0.3s cubic-bezier(0.455,0.03,0.515,0.955);
  -webkit-transition: opacity 0.3s cubic-bezier(0.455,0.03,0.515,0.955), transform 0.3s cubic-bezier(0.455,0.03,0.515,0.955);
  transition: opacity 0.3s cubic-bezier(0.455,0.03,0.515,0.955), transform 0.3s cubic-bezier(0.455,0.03,0.515,0.955);
  -webkit-transform: scale(0.7);
  -ms-transform: scale(0.7);
  transform: scale(0.7);
}

.c0.open-enter-active,
.c0.open-enter-active:before {
  -webkit-transition: opacity 0.3s cubic-bezier(0.455,0.03,0.515,0.955);
  transition: opacity 0.3s cubic-bezier(0.455,0.03,0.515,0.955);
  display: block;
  opacity: 1;
}

.c0.open-enter-active {
  -webkit-transition: opacity 0.3s cubic-bezier(0.455,0.03,0.515,0.955), -webkit-transform 0.3s cubic-bezier(0.455,0.03,0.515,0.955);
  -webkit-transition: opacity 0.3s cubic-bezier(0.455,0.03,0.515,0.955), transform 0.3s cubic-bezier(0.455,0.03,0.515,0.955);
  transition: opacity 0.3s cubic-bezier(0.455,0.03,0.515,0.955), transform 0.3s cubic-bezier(0.455,0.03,0.515,0.955);
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.c0.open-enter-done,
.c0.open-enter-active:before {
  -webkit-transition: opacity 0.3s cubic-bezier(0.455,0.03,0.515,0.955);
  transition: opacity 0.3s cubic-bezier(0.455,0.03,0.515,0.955);
  display: block;
  opacity: 1;
}

.c0.open-enter-done {
  -webkit-transition: opacity 0.3s cubic-bezier(0.455,0.03,0.515,0.955), -webkit-transform 0.3s cubic-bezier(0.455,0.03,0.515,0.955);
  -webkit-transition: opacity 0.3s cubic-bezier(0.455,0.03,0.515,0.955), transform 0.3s cubic-bezier(0.455,0.03,0.515,0.955);
  transition: opacity 0.3s cubic-bezier(0.455,0.03,0.515,0.955), transform 0.3s cubic-bezier(0.455,0.03,0.515,0.955);
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.c0.open-exit,
.c0.open-exit:before {
  display: block;
  opacity: 1;
}

.c0.open-exit {
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.c0.open-exit-active,
.c0.open-exit-active:before {
  display: block;
  opacity: 0;
}

.c0.open-exit-active {
  -webkit-transform: scale(0.7);
  -ms-transform: scale(0.7);
  transform: scale(0.7);
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 16px;
  border-bottom: none;
  line-height: 1;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  text-align: center;
}

.c2:first-child {
  text-align: left;
}

.c2:last-child {
  text-align: right;
}

.c3 {
  font-size: 14px;
  text-transform: uppercase;
  margin-bottom: 8px;
  color: #999999;
}

.c4 {
  font-size: 16px;
  font-weight: 600;
  color: #262626;
}

<div
  className="c0"
  direction="bottom"
  style={
    Object {
      "padding": 0,
      "width": "260px",
    }
  }
>
  <div
    className="c1"
    size="large"
  >
    <div
      className="c2"
    >
      <div
        className="c3"
        size="large"
      >
        Sec
      </div>
      <div
        className="c4"
        size="large"
      >
        
      </div>
    </div>
    <div
      className="c2"
    >
      <div
        className="c3"
        size="large"
      >
        Row
      </div>
      <div
        className="c4"
        size="large"
      >
        5
      </div>
    </div>
    <div
      className="c2"
    >
      <div
        className="c3"
        size="large"
      >
        Seat
      </div>
      <div
        className="c4"
        size="large"
      >
        25
      </div>
    </div>
  </div>
</div>
`;

exports[`SeatTooltip should match snapshot when small and dark 1`] = `
.c0 {
  background-color: #1f262d;
  border: 1px solid #1f262d;
  border-radius: 4px;
  box-shadow: 0 4px 4px 0 rgba(0,0,0,0.12);
  position: absolute;
  max-width: 260px;
  color: rgba(255,255,255,1);
  padding: 8px;
  display: none;
  font-size: 12px;
  -webkit-transition: opacity 0.1s cubic-bezier(0.55,0.085,0.68,0.53), -webkit-transform 0.1s cubic-bezier(0.55,0.085,0.68,0.53);
  -webkit-transition: opacity 0.1s cubic-bezier(0.55,0.085,0.68,0.53), transform 0.1s cubic-bezier(0.55,0.085,0.68,0.53);
  transition: opacity 0.1s cubic-bezier(0.55,0.085,0.68,0.53), transform 0.1s cubic-bezier(0.55,0.085,0.68,0.53);
}

.c0:before {
  content: "";
  position: absolute;
  -webkit-transition: opacity 0.1s cubic-bezier(0.55,0.085,0.68,0.53), scale 0.1s cubic-bezier(0.55,0.085,0.68,0.53);
  transition: opacity 0.1s cubic-bezier(0.55,0.085,0.68,0.53), scale 0.1s cubic-bezier(0.55,0.085,0.68,0.53);
  display: none;
  border-top-left-radius: 100%;
  width: 12px;
  height: 12px;
  -webkit-transform: translateY(-50%) rotate(-135deg);
  -ms-transform: translateY(-50%) rotate(-135deg);
  transform: translateY(-50%) rotate(-135deg);
  background-color: #1f262d;
  left: calc(50% - 6px);
  top: -1px;
  -webkit-transform: translateY(-50%) rotate(-135deg);
  -ms-transform: translateY(-50%) rotate(-135deg);
  transform: translateY(-50%) rotate(-135deg);
}

.c0.open-enter,
.c0.open-enter:before {
  -webkit-transition: opacity 0.3s cubic-bezier(0.455,0.03,0.515,0.955);
  transition: opacity 0.3s cubic-bezier(0.455,0.03,0.515,0.955);
  display: block;
  opacity: 0;
}

.c0.open-enter {
  -webkit-transition: opacity 0.3s cubic-bezier(0.455,0.03,0.515,0.955), -webkit-transform 0.3s cubic-bezier(0.455,0.03,0.515,0.955);
  -webkit-transition: opacity 0.3s cubic-bezier(0.455,0.03,0.515,0.955), transform 0.3s cubic-bezier(0.455,0.03,0.515,0.955);
  transition: opacity 0.3s cubic-bezier(0.455,0.03,0.515,0.955), transform 0.3s cubic-bezier(0.455,0.03,0.515,0.955);
  -webkit-transform: scale(0.7);
  -ms-transform: scale(0.7);
  transform: scale(0.7);
}

.c0.open-enter-active,
.c0.open-enter-active:before {
  -webkit-transition: opacity 0.3s cubic-bezier(0.455,0.03,0.515,0.955);
  transition: opacity 0.3s cubic-bezier(0.455,0.03,0.515,0.955);
  display: block;
  opacity: 1;
}

.c0.open-enter-active {
  -webkit-transition: opacity 0.3s cubic-bezier(0.455,0.03,0.515,0.955), -webkit-transform 0.3s cubic-bezier(0.455,0.03,0.515,0.955);
  -webkit-transition: opacity 0.3s cubic-bezier(0.455,0.03,0.515,0.955), transform 0.3s cubic-bezier(0.455,0.03,0.515,0.955);
  transition: opacity 0.3s cubic-bezier(0.455,0.03,0.515,0.955), transform 0.3s cubic-bezier(0.455,0.03,0.515,0.955);
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.c0.open-enter-done,
.c0.open-enter-active:before {
  -webkit-transition: opacity 0.3s cubic-bezier(0.455,0.03,0.515,0.955);
  transition: opacity 0.3s cubic-bezier(0.455,0.03,0.515,0.955);
  display: block;
  opacity: 1;
}

.c0.open-enter-done {
  -webkit-transition: opacity 0.3s cubic-bezier(0.455,0.03,0.515,0.955), -webkit-transform 0.3s cubic-bezier(0.455,0.03,0.515,0.955);
  -webkit-transition: opacity 0.3s cubic-bezier(0.455,0.03,0.515,0.955), transform 0.3s cubic-bezier(0.455,0.03,0.515,0.955);
  transition: opacity 0.3s cubic-bezier(0.455,0.03,0.515,0.955), transform 0.3s cubic-bezier(0.455,0.03,0.515,0.955);
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.c0.open-exit,
.c0.open-exit:before {
  display: block;
  opacity: 1;
}

.c0.open-exit {
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.c0.open-exit-active,
.c0.open-exit-active:before {
  display: block;
  opacity: 0;
}

.c0.open-exit-active {
  -webkit-transform: scale(0.7);
  -ms-transform: scale(0.7);
  transform: scale(0.7);
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 8px;
  border-bottom: 1px solid #999999;
  line-height: 1;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  text-align: center;
}

.c2:first-child {
  text-align: left;
}

.c2:last-child {
  text-align: right;
}

.c3 {
  font-size: 12px;
  text-transform: uppercase;
  margin-bottom: 8px;
  color: #999999;
}

.c4 {
  font-size: 14px;
  font-weight: 600;
  color: rgba(255,255,255,1);
}

.c5 {
  padding: 8px;
  font-size: 12px;
  color: rgba(255,255,255,1);
}

<div
  className="c0"
  direction="bottom"
  style={
    Object {
      "padding": 0,
      "width": "180px",
    }
  }
>
  <div
    className="c1"
    size="small"
  >
    <div
      className="c2"
    >
      <div
        className="c3"
        size="small"
      >
        Sec
      </div>
      <div
        className="c4"
        size="small"
      >
        B
      </div>
    </div>
    <div
      className="c2"
    >
      <div
        className="c3"
        size="small"
      >
        Row
      </div>
      <div
        className="c4"
        size="small"
      >
        5
      </div>
    </div>
    <div
      className="c2"
    >
      <div
        className="c3"
        size="small"
      >
        Seat
      </div>
      <div
        className="c4"
        size="small"
      >
        25
      </div>
    </div>
  </div>
  <div
    className="c5"
    size="small"
  >
    <div>
      test
    </div>
  </div>
</div>
`;
